rocAL/rocAL/source/coco_meta_data_reader_key_points.cpp

```diff
@@ -0,0 +1,388 @@
+            while (parser.NextArrayValue())
+            {
+                if (parser.PeekType() != kObjectType)
+                {
+                    continue;
+                }
+                parser.EnterObject();
+                while (const char *internal_key = parser.NextObjectKey())
+                {
+                    if (0 == std::strcmp(internal_key, "id"))
+                    {
+                        id = parser.GetInt();
+                    }
+                    else
+                    {
+                        parser.SkipValue();
+                    }
+                }
+                _label_info.insert(std::make_pair(id, continuous_idx));
+                continuous_idx++;
+            }
+        }
+        else if (0 == std::strcmp(key, "annotations"))
+        {
+            RAPIDJSON_ASSERT(parser.PeekType() == kArrayType);
+            parser.EnterArray();
+            while (parser.NextArrayValue())
+            {
+                int id = 1, label = 0, is_crowd = 0;
+                float joint_sum = 0.0, area = 0.0;
+                long int ann_id = 0;
+                std::array<float, NUMBER_OF_JOINTS * 3> keypoint{};
+                if (parser.PeekType() != kObjectType)
+                {
+                    continue;
+                }
+                parser.EnterObject();
+                while (const char *internal_key = parser.NextObjectKey())
+                {
+                    if (0 == std::strcmp(internal_key, "image_id"))
+                    {
+                        id = parser.GetInt();
+                    }
+                    else if (0 == std::strcmp(internal_key, "category_id"))
+                    {
+                        label = parser.GetInt();
+                    }
+                    else if (0 == std::strcmp(internal_key, "id"))
+                    {
+                        ann_id = parser.GetDouble();
+                    }
+                    else if (0 == std::strcmp(internal_key, "is_crowd"))
+                    {
+                        is_crowd = parser.GetInt();
+                    }
+                    else if (0 == std::strcmp(internal_key, "area"))
+                    {
+                        area = parser.GetDouble();
+                    }
+                    else if (0 == std::strcmp(internal_key, "bbox"))
+                    {
+                        RAPIDJSON_ASSERT(parser.PeekType() == kArrayType);
+                        parser.EnterArray();
+
+                        box_center.push_back(parser.NextArrayValue() * parser.GetDouble());
+                        box_center.push_back(parser.NextArrayValue() * parser.GetDouble());
+                        box_scale.push_back(parser.NextArrayValue() * parser.GetDouble());
+                        box_scale.push_back(parser.NextArrayValue() * parser.GetDouble());
+
+                        // Move to next section
+                        parser.NextArrayValue();
```

Comment:
Looks like this code will break the cases with multiple bboxes. Please check if this handles that case

Reply:
HRNet pipeline processes each annotation as a separate image. This case is handled

Reply:
Not sure how that works in parsing json file. Did you mean the json file is modified by HRNet pipeline?